cf-custom-resources/lib/env-controller.js

```diff
@@ -232,16 +261,43 @@ exports.handler = async function (event, context) {
 
 function setOfParameterKeysWithWorkload(cfnParams, workload) {
   const envSet = new Set();
-  cfnParams.forEach(param => {
-    var values = new Set(param.ParameterValue.split(','));
-    if (!values.has(workload)) {
-      return;
+  cfnParams.forEach((param) => {
+    if (param.ParameterKey !== AliasParamKey) {
+      let values = new Set(param.ParameterValue.split(","));
+      if (!values.has(workload)) {
+        return;
+      }
     }
     envSet.add(param.ParameterKey);
-  })
-  return envSet
+  });
+  return envSet;
+}
+
+function needUpdateAliases(cfnParams, workload, aliases) {
+  let requireUpdate = false;
+  cfnParams.forEach((param) => {
+    if (param.ParameterKey !== AliasParamKey) {
+      return;
+    }
+    let obj = JSON.parse(param.ParameterValue || "{}");
+    if ((obj[workload] || []).toString() !== aliases.toString()) {
+      requireUpdate = true;
+    }
+  });
+  return requireUpdate;
 }
 
+const updateAliases = function (workload, aliases, cfnAliases) {
```

Comment:
nit: to keep the param order consistent between `needUpdateAliases` and this fn can we reorder like this:
```suggestion
const updateAliases = function (cfnAliases, workload, aliases) {
```